.github/workflows/build.yml

```diff
@@ -103,7 +103,8 @@ jobs:
 
     - name: Record dependencies
       id: record-deps
-      run: cabal build all --dry-run
+      run: |
+        cabal build all --dry-run
```

Comment:
Why did this not make CI fail?

https://github.com/input-output-hk/ouroboros-network/actions/runs/4084241519/jobs/7040712757#step:11:13

Reply:
Tbh I have no idea. Perhaps Cabal doesn't end with an exit code that is recognized as failing in this case?